test/league_stat_test.rb

```diff
@@ -17,6 +17,78 @@ def test_it_exists
     assert_instance_of LeagueStat, @league_stat
   end
 
+  def test_it_is_a_hash
+    assert Hash, @league_stat.class
+  end
+
+  def test_it_nests_second_hash_with_default_value_0
+    assert_equal 0, @league_stat.stats_by_team[:new_key][:doesnt_exist]
+  end
+
+  def test_it_creates_teams
+    test_team = OpenStruct.new(team_id:100, team_name:"TestTeam")
+    @league_stat.create_teams([test_team])
+
+    assert_equal "TestTeam", @league_stat.stats_by_team[100][:team_name]
+  end
```

Comment:
i used openstructs for these. they allow a hash to be called on as though it were an object, not a hash. you can hash.key to return value instead of hash[key]. it lets me mimic a game or team object without initializing and instantiating a real one.

Reply:
Nice find!